ledger.rb

```diff
@@ -1,24 +1,156 @@
   def balance
-    parser = Parser.new
-    parsed_info = parser.parse_ledger("index.ledger")
+    transactions = {}
+    transactions_sums = {}
+
+    @parser.parsed_file.each do |transaction|
+      transaction[:accounts].each do |account|
+        if @balances.include?(account[:currency])
+          @balances[account[:currency]] += account[:amount]
+        else
+          @balances[account[:currency]] = account[:amount]
+        end
+
+        @balances[account[:currency]] = @balances[account[:currency]].round(2)
+
+        if transactions_sums.include?(account[:description])
+          transactions_sums[account[:description]] += account[:amount]
+          transactions[account[:description]] = full_action(transactions_sums[account[:description]], account[:currency])
+        else
+          transactions_sums[account[:description]] = account[:amount]
+          transactions[account[:description]] = full_action(account[:amount], account[:currency])
+        end
+      end
+    end
+
+    transactions.sort_by { |k, v| v }.reverse.each do |description, action|
+      balance_line(action, description)
+    end
+
+    puts '--------------------'
+
+    @balances.each do |k, v|
+      balance_line(full_action(v, k))
+    end
 
-    puts parsed_info
   end
 
   desc "print", "The print command prints out ledger transactions in a textual format that can be parsed by Ledger."
   def print
-    puts "print command"
+    @parser.parsed_file.each do |transaction|
+      print_title_line(transaction[:date], transaction[:description])
+
+      transaction[:accounts].each do |account|
+        print_line(account[:description], account[:amount], account[:currency])
+      end
+
+      puts "\n"
+    end
+
+  end
+
+  private
+
+  def full_action(amount, currency)
+    "#{'%.2f' % amount} #{currency}"
+  end
+
+  def balance_line(action, description = nil)
+    action = " " * (20 - action.size) + action + ' '
+    action = action.red if action.match(/-/)
+    blue_desc = "#{description}".blue
+
+    puts action + blue_desc
+  end
+
+  def register_title_line(date, description)
+    puts "#{date}".light_black + " #{description} ".light_white
   end
+
+  def register_line(description, amount, balances, currency)
+    action = full_action(amount, currency)
+    balance_text = register_balance_text(balances)
+    blue_desc = "    #{description}".blue
+    desc_space = ' ' * (50 - description.size) + ' ' * (20 - action.size)
+    balance_space = ' ' * (30 - balance_text[/\w/].size)
+    amount_full_action = "#{action}"
+    amount_full_action = amount_full_action.red if amount < 0
+
+    puts blue_desc + desc_space + amount_full_action + balance_space + balance_text
+  end
+
+  def register_balance(balances)
+    balance_text = register_balance_text(balances)
+
+    puts ' ' * 103 + '-' * 23
+    puts ' ' * 103 + balance_text
+  end
+
+  def register_balance_text(balances)
+    balance_text = ''
+
+    balances.each_with_index do |b, i|
```

Comment:
let's use the complete name with these variables, it would be easier to read

```suggestion
    balances.each_with_index do |balance, index|
```